tests/storage_test.go

```diff
@@ -67,71 +68,99 @@ var _ = Describe("[Serial]Storage", func() {
 
-				nfsInitialized = true
-			}
-			return _pvName
+		initNFS := func() *k8sv1.Pod {
+			tests.SkipNFSTestIfRunnigOnKindInfra()
+
+			// Prepare a NFS backed PV
+			By("Starting an NFS POD")
+			os := string(cd.ContainerDiskAlpine)
+			nfsPod := tests.CreateNFSTargetPOD(os)
+
+			return nfsPod
 		}
 
-		BeforeEach(func() {
-			nfsInitialized = false
-		})
+		createNFSPvAndPvc := func(ipFamily k8sv1.IPFamily, nfsPod *k8sv1.Pod) string {
+			pvName := "test-nfs" + rand.String(48)
+
+			// create a new PV and PVC (PVs can't be reused)
+			By("create a new NFS PV and PVC")
+			nfsIP := libnet.GetPodIpByFamily(nfsPod, ipFamily)
+			ExpectWithOffset(1, nfsIP).NotTo(BeEmpty())
+			os := string(cd.ContainerDiskAlpine)
+			tests.CreateNFSPvAndPvc(pvName, "5Gi", nfsIP, os)
+			return pvName
+		}
 
-		AfterEach(func() {
-			if nfsInitialized {
-				By("Deleting the VMI")
-				Expect(virtClient.VirtualMachineInstance(vmi.Namespace).Delete(vmi.Name, &metav1.DeleteOptions{})).To(Succeed())
-
-				By("Waiting for VMI to disappear")
-				tests.WaitForVirtualMachineToDisappearWithTimeout(vmi, 120)
-				// PVs can't be reused
-				tests.DeletePvAndPvc(_pvName)
-
-				By("Deleting NFS pod")
-				Expect(virtClient.CoreV1().Pods(tests.NamespaceTestDefault).Delete(tests.NFSTargetName, &metav1.DeleteOptions{})).To(Succeed())
-				By("Waiting for NFS pod to disappear")
-				tests.WaitForPodToDisappearWithTimeout(tests.NFSTargetName, 120)
-			}
-		})
 		Context("[rfe_id:3106][crit:medium][vendor:cnv-qe@redhat.com][level:component]with Alpine PVC", func() {
-			table.DescribeTable("should be successfully started", func(newVMI VMICreationFunc, storageEngine string) {
-				tests.SkipPVCTestIfRunnigOnKindInfra()
 
-				var ignoreWarnings bool
+			Context("should be successfully", func() {
 				var pvName string
-				// Start the VirtualMachineInstance with the PVC attached
-				if storageEngine == "nfs" {
-					pvName = initNFS()
-					ignoreWarnings = true
-				} else {
-					pvName = tests.DiskAlpineHostPath
-				}
-				vmi = newVMI(pvName)
-				tests.RunVMIAndExpectLaunchWithIgnoreWarningArg(vmi, 120, ignoreWarnings)
+				var nfsPod *k8sv1.Pod
 
-				By("Checking that the VirtualMachineInstance console has expected output")
-				expecter, err := tests.LoggedInAlpineExpecter(vmi)
-				Expect(err).ToNot(HaveOccurred())
-				expecter.Close()
-			},
-				table.Entry("[test_id:3130]with Disk PVC", tests.NewRandomVMIWithPVC, ""),
-				table.Entry("[test_id:3131]with CDRom PVC", tests.NewRandomVMIWithCDRom, ""),
-				table.Entry("[test_id:4618]with NFS Disk PVC", tests.NewRandomVMIWithPVC, "nfs"),
-			)
+				AfterEach(func() {
+					if vmi != nil {
+						By("Deleting the VMI")
+						Expect(virtClient.VirtualMachineInstance(vmi.Namespace).Delete(vmi.Name, &metav1.DeleteOptions{})).To(Succeed())
+
+						By("Waiting for VMI to disappear")
+						tests.WaitForVirtualMachineToDisappearWithTimeout(vmi, 120)
+
+						vmi = nil
```

Comment:
a note
if the `Expect` will assert, we wont reach this line,
the next cycle can have the vmi which isnt nil of the previous round in case it didnt set it,
but at the worst case the `Expect` will catch again, so basically no harm done, 
just need to know about it
or even better to add a defer function in the "func" that starts on line 102

same issue in couple places.

Reply:
I believe that as part of the review process of this PR the cleanup of the resources was over optimized and over complicated.  I don't will to add any more complication to it.

Reply:
I dont see it was over X, and a little defer function would just make it better easily in this case.
Its not the first PR we did it.
up to you, but i think otherwise.